x-pack/platform/plugins/shared/streams/server/routes/streams/ingest/route.ts

```diff
@@ -84,16 +84,20 @@ const upsertIngestRoute = createServerRoute({
     const { streamsClient, assetClient } = await getScopedClients({
       request,
     });
+    const name = params.path.name;
+    const existingStream = await streamsClient.getStream(name).catch(() => undefined);
 
     if (
-      isWiredStreamDefinition({ name: params.path.name, ...params.body }) &&
+      isWiredStreamDefinition({
```

Comment:
preferably we'd do `isWiredStreamUpsertRequest`, but maybe we can avoid it entirely? can we use `asStreamUpsertRequest` below?

Reply:
don't we need to check if streams are disable in case of a wired ingest upsert? I'm adding a `isWiredIngestUpsertRequest` instead of relying on `isWiredStreamDefinition`